x/superfluid/keeper/grpc_query.go

```diff
@@ -278,6 +279,78 @@ func (q Querier) SuperfluidDelegationsByDelegator(goCtx context.Context, req *ty
 	return &res, nil
 }
 
+// UserSuperfluidPositionsPerConcentratedPoolBreakdown returns all the cl superfluid positions for the specified delegator in the specified concentrated liquidity pool.
+func (q Querier) UserSuperfluidPositionsPerConcentratedPoolBreakdown(goCtx context.Context, req *types.UserSuperfluidPositionsPerConcentratedPoolBreakdownRequest) (*types.UserSuperfluidPositionsPerConcentratedPoolBreakdownResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	delAddr, err := sdk.AccAddressFromBech32(req.DelegatorAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the position IDs for the specified pool ID and user address.
+	positions, err := q.Keeper.clk.GetUserPositions(ctx, delAddr, req.ConcentratedPoolId)
+	if err != nil {
+		return nil, err
+	}
+
+	// Query each position ID and determine if it has a lock ID associated with it, which implies the position is superfluid staked.
+	// Construct a response with the position ID, lock ID, the amount of cl shares staked, and what those shares are worth in staked osmo tokens.
+	var clPoolUserPositionRecords []types.ConcentratedPoolUserPositionRecord
+	for _, pos := range positions {
+		lockId, err := q.Keeper.clk.GetLockIdFromPositionId(ctx, pos.PositionId)
+		switch err.(type) {
+		case cltypes.PositionIdToLockNotFoundError:
+			continue
+		case nil:
+			if lockId == 0 {
+				continue
+			}
```

Comment:
If this is a `nil` case and we get a zero `lockId`, wouldn't this be an error?

Reply:
You are correct, was just a defense in depth in the event `GetLockIdFromPositionId` was extended in some way. Removed the extra check here https://github.com/osmosis-labs/osmosis/pull/5600/commits/8ca6a5dfae284df651fb4b3db4cf162777c49ac1